File: exercises/day13-Keyboard-First-Refactoring/kotlin/src/main/kotlin/sleightrip/exception/Exceptions.kt
package sleightrip.exception

/**
 * Thrown when no elf is logged into the workshop system.
 */
class ElfNotLoggedInException : RuntimeException("No elf is logged into the workshop system")

/**
 * Thrown when a static collaborator (like UserSession or TripDAO) is called directly.
 */
class CollaboratorCallException(message: String) : RuntimeException(message)

File: exercises/day13-Keyboard-First-Refactoring/csharp/SleighTripService/Exception/Exceptions.cs
namespace SleighTripService.Exception;

/// <summary>
/// Thrown when no elf is logged into the workshop system.
/// </summary>
public class ElfNotLoggedInException : System.Exception
{
    public ElfNotLoggedInException() 
        : base("No elf is logged into the workshop system")
    {
    }
}

/// <summary>
/// Thrown when a static collaborator (like UserSession or TripDAO) is called directly.
/// </summary>
public class CollaboratorCallException : System.Exception
{
    public CollaboratorCallException(string message) : base(message)
    {
    }
}
